### PR TITLE
Stop building with OpenSSL 1.1 and libcurl 7.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ source:
     - 0001-remove-wil-from-exported-config.patch
 
 build:
-  number: 0
+  number: 1
```
Comment (jdblischak) [inline recipe/meta.yaml]: I don't think there is any need to upload new binaries for this change. We are simply dropping build variants. Bumping the build number to 1 will cause an identical libcurl8/openssl3 variant to be uploaded to anaconda.org

**Review by jdblischak — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (teo-tsirpanis) [reply recipe/meta.yaml]: Reverted.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,11 +0,0 @@
-openssl:
-- '1.1.1'
-- '3'
-- '3'
-libcurl:
-- 7
-- 7
```
Comment (jjerphan) [inline recipe/conda_build_config.yaml:7]: I think the title of the PR must be changed to include the removal of libcurl 7.

**Review by jjerphan — CHANGES_REQUESTED** [review]
I just have one comment.

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (teo-tsirpanis) [reply recipe/conda_build_config.yaml:7]: Updated title.

**Review by jjerphan — APPROVED** [review]

**Review by jdblischak — APPROVED** [review]

**Pull request merged**